ksonnet-util/util.libsonnet

```diff
@@ -174,7 +174,8 @@ local util(k) = {
 
     deployment.mapContainers(addMount) +
     deployment.mixin.spec.template.spec.withVolumesMixin([
-      volume.fromHostPath(name, hostPath),
+      volume.fromHostPath(name, hostPath) +
+      if type != '' then volume.hostPath.withType(type) else {},
```

Comment:
Given that empty string is the default, would it make sense to remove the conditional logic?

Maybe the question is, if we make that change, does that generated a lot of diff and is something that we are trying to prevent?

Reply:
Ah yeah, I could have said - I tried it like that and a lot of my call sites ended up having `type: ""` added :(

Reply:
Indeed I think we need to keep the logic, without assuming implementation's default while also not polluting with "nil" values.

Reply:
@iainlane and I jumped on a call before and my suggestion was to provide a new mixin instead. Something like:

```
hostVolumeMount(name, hostPath, path, readOnly=false, volumeMountMixin={}, volumeMixin={})::
(...)
```

That gives the caller more flexibility, and we are already doing it for the volumeMount